xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/table/livetabledate.js

```diff
@@ -63,6 +63,20 @@ require(['jquery', 'jdateformatparser', 'daterangepicker', 'xwiki-events-bridge'
           updateInput(this, event, picker);
         });
 
+        input.on('keypress', function (e) {
+          if(e.which !== 13) {
```

Comment:
It was OK to just wrap the ``return`` in a block, but if you want to wrap the code then you need to update the condition . Right now this code is executed for any key except Enter.